### PR TITLE
Add general cleanup class to CI to prevent FS resources exhaustion

### PR DESCRIPTION
**Pull request by rtrinque**
This add a daily run script that get rid of workspaces in jenkins
that are older than jenkins_worspace_max_age.
Over time the accumulation of files in the workspaces can lead to
inodes or space exhaustion on the filesystem. This script should
keep this under control.

Trello: https://trello.com/c/JwTIJsk1/2920-stop-low-available-disk-inodes-alert-3

### CODE REVIEW

File: modules/govuk_ci/manifests/cleanup.pp [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+# [*jenkins_workspace_max_age*]
+#   Maximum age -in days- of Jenkins workspaces before they are deleted by the cleanup job.
+#
+class govuk_ci::cleanup (
+  $jenkins_workspace_max_age = 90,
+){
+  cron::crondotdee { 'clean_up_workspace':
+    command => 'find /var/lib/jenkins/workspace/ -maxdepth 1 -mindepth 1 -type d -mtime +$jenkins_workspace_max_age -exec rm -rf {} \;',
+    hour    => 7,
+    minute  => 25,
```
Comment (ChrisBAshton) [inline modules/govuk_ci/manifests/cleanup.pp:16]: Any particular reason for this time of day?

**Review by ChrisBAshton — APPROVED** [review]
LGTM 👍 ran the command on all environments (without the `rm -rf`) and played around with the ages, looks like it should clear up a lot of files.

File: modules/govuk_ci/manifests/cleanup.pp (hunk 1 above)
Comment (rtrinque) [reply modules/govuk_ci/manifests/cleanup.pp:16]: There already a cleaning job running a little later, so just offset it to a little earlier

File: hieradata_aws/class/integration/ci_agent.yaml [hunk 2]
```diff
@@ -7,6 +7,7 @@ mongodb::server::replicaset_members:
 
 clamav::use_service: false
 
+govuk_ci::agent::cleanup::jenkins_workspace_max_age: '30'
```
Comment (ChrisBAshton) [inline hieradata_aws/class/integration/ci_agent.yaml:10]: 👍 thanks

**Pull request merged**